src/server/server.cc

```diff
@@ -2115,3 +2122,40 @@ AuthResult Server::AuthenticateUser(const std::string &user_password, std::strin
   *ns = kDefaultNamespace;
   return AuthResult::IS_ADMIN;
 }
+
+void Server::evictionClients() {
+  size_t mem = 0;
+  for (const auto &t : worker_threads_) {
+    mem += t->GetWorker()->GetConnectionsMemoryUsed();
+  }
+  if (mem < config_->max_memory_clients) {
+    return;
+  }
+
+  std::vector<redis::Connection *> conns;
+  for (const auto &t : worker_threads_) {
+    auto worker_conns = t->GetWorker()->GetConnections();
+    for (const auto &iter : worker_conns) {
+      conns.push_back(iter.second);
```

Comment:
Are these thread-safe? Is it possible that the connections has been released?

Reply:
I referred to the implementation of the `void Server::decreaseWorkerThreads(size_t delta)` function, which is also called without locking.
https://github.com/AntiTopQuark/kvrocks/blob/43f14db05f73d02a6fefc95347fd5c79781f85a5/src/server/server.cc#L1895

Reply:
@git-hulk I think there's something wrong here. Am I missing something?

Reply:
In `MigrateConnection`, the connection cannot be released.

Reply:
The lock will not work here, and these connections may still be released when you access them below.

One of my own practices is to `maxmemory-clients/worker`, estimate the `maxmemory` of each worker, and then record (update in real time) the memory of each connection in each worker, which can be more timely and without locking.

Reply:
I referred to the implementation of the KickoutIdleClients() function. In FreeConnectionByID(), if the connection (conn) has already been released, no further action will be taken.

Additionally, may I ask if I can assume that, in most cases, the memory usage across different workers in the system is almost uniform, and that there is rarely a situation where memory usage is uneven?

Reply:
Yes, we usually assume that the worker is balanced. In fact, I do this, it is just a trade-off